MapleServer2/Managers/TrophyManager.cs

```diff
@@ -42,13 +42,13 @@ public static void OnPlayTimeTick(Player player)
 
     public static void OnLevelUp(Player player)
     {
-        int jobId = (int) player.JobCode;
+        int jobId = (int) player.Job;
+        int level = player.Levels.Level;
         IEnumerable<TrophyMetadata> levelUpTrophies = GetRelevantTrophies(TrophyTypes.LevelUp);
         IEnumerable<TrophyMetadata> levelTrophies = GetRelevantTrophies(TrophyTypes.Level);
 
-        // TODO: This might not be right :D
         IEnumerable<TrophyMetadata> matchingTrophies = levelUpTrophies
-            .Where(t => IsMatchingCondition(t.Grades.First().ConditionCodes, jobId));
+            .Where(t => IsMatching(t.Grades.First().ConditionCodes, jobId) && IsMatching(t.Grades.First().ConditionTargets, level));
```

Comment:
Have we checked the trophies for the progress steps to see that they only have 1? I expect "Reach X level on Y job" to have only 1 progress step, so this will only add 1 to progress upon hitting the right level, but who knows with how crazy this game is sometimes 🤷🏻‍♂️